libraries/chain/include/eosio/chain/exceptions.hpp

```diff
@@ -634,8 +630,21 @@ namespace eosio { namespace chain {
                                  3250000, "Protocol feature exception" )
       FC_DECLARE_DERIVED_EXCEPTION( protocol_feature_validation_exception, protocol_feature_exception,
                                     3250001, "Protocol feature validation exception" )
-      FC_DECLARE_DERIVED_EXCEPTION( protocol_feature_bad_block_exception, protocol_feature_exception,
-                                    3250002, "Protocol feature exception (invalid block)" )
       FC_DECLARE_DERIVED_EXCEPTION( protocol_feature_iterator_exception, protocol_feature_exception,
                                     3250003, "Protocol feature iterator exception" )
+
+   // Any derived types of subjective_block_production_exception need to update controller::failure_is_subjective
+   FC_DECLARE_DERIVED_EXCEPTION( subjective_block_production_exception,  chain_exception,
+                                 3260000, "Subjective exception thrown during block production" )
+
+   // Objective block production exceptions are throw out of transaction processing to stop block production
+   FC_DECLARE_DERIVED_EXCEPTION( objective_block_production_exception,  chain_exception,
+                                 3270000, "Objective exception thrown during block production" )
```

Comment:
I would say "block validation" everywhere you currently say "block production".

Reply:
Updated